### PR TITLE
feat: v13 `Webhooks`.

### PR DESCRIPTION
**Pull request by WiseDevHelper**
**Please describe the changes this PR makes and why it should be merged:**

[Demo](https://deploy-preview-700--discordjs-guide.netlify.app/popular-topics/webhooks.html)

This PR updates the `Webhooks` section of the guide for the `v13` release.

**Status**
- [x] Update `Webhooks`.
- [x] Update code samples.

Please let me know if there is more to change. I'll be happy to work on them.

### CODE REVIEW

**Comment by WiseDevHelper** [comment]
Should I replace the code for the parts where we have examples for `Webhook` and `WebhookClient` with only the essential corresponding function like the previous parts?

File: guide/popular-topics/webhooks.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
 ## What is a webhook
 
-Webhooks are a utility used to send messages to text channels without needing a Discord application. Webhooks are useful for allowing something to send messages without requiring a Discord application. However, you should note that you cannot directly edit or delete messages you sent through the webhook. discord.js introduces two structures to make use of this functionality, `Webhook` and `WebhookClient`. `WebhookClient` is an extended version of a `Webhook` which allows you to send messages through it without needing a bot client.
+Webhooks are a utility used to send messages to text channels without needing a Discord application. Webhooks are useful for allowing something to send messages without requiring a Discord application. You can also directly edit or delete messages you sent through the webhook. Discord.js introduces two structures to make use of this functionality, `Webhook` and `WebhookClient`. `WebhookClient` is an extended version of a `Webhook` which allows you to send messages through it without needing a bot client.
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:7]:
```suggestion
Webhooks are a utility used to send messages to text channels without needing a Discord application. Webhooks are useful for allowing something to send messages without requiring a Discord application. You can also directly edit or delete messages you sent through the webhook. There are two structures to make use of this functionality: `Webhook` and `WebhookClient`. `WebhookClient` is an extended version of a `Webhook`, which allows you to send messages through it without needing a bot client.
```

File: guide/popular-topics/webhooks.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # Webhooks
 
-Webhooks can send messages to a text channel without having to log in as a bot. discord.js implements a variety of methods to fetch, create, edit, and use webhooks. In this section, you will learn how to create, fetch, edit, and use webhooks.
+Webhooks can send messages to a text channel, and fetch, edit, delete the messages sent by them without having to log in as a bot. Discord.js implements a variety of methods to fetch, create, edit, and use webhooks. In this section, you will learn how to create, fetch, edit, and use webhooks.
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:3]:
```suggestion
Webhooks can send messages to a text channel without having to log in as a bot. They can also fetch, edit, and delete their own messages. There are a variety of methods in discord.js to interact with webhooks. In this section, you will learn how to create, fetch, edit, and use webhooks.
```

File: guide/popular-topics/webhooks.md [hunk 3]
```diff
@@ -67,8 +67,7 @@ Once you are there, click on the `Create Webhook` / `New Webhook` button; this w
 
 ### Creating webhooks with discord.js
 
-
-discord.js provides a method for creating webhooks called <DocsLink path="class/TextChannel?scrollTo=createWebhook">`TextChannel#createWebhook()`</DocsLink>.
+Discord.js provides a method for creating webhooks called <DocsLink path="class/TextChannel?scrollTo=createWebhook">`TextChannel#createWebhook()`</DocsLink>.
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:70]:
```suggestion
Webhooks can be created with the <DocsLink path="class/TextChannel?scrollTo=createWebhook">`TextChannel#createWebhook()`</DocsLink> method.
```

File: guide/popular-topics/webhooks.md [hunk 4]
```diff
@@ -94,21 +93,26 @@ webhook.edit({
 
 ## Using webhooks
 
-Webhooks, unlike bots, can send more than one embed per message, up to 10. They can also send attachments and normal content. The <DocsLink path="class/Webhook?scrollTo=send">`Webhook#send()`</DocsLink> method to send to a webhook is very similar to the method for sending to a text channel. Webhooks can also choose how the username and avatar will appear when they send the message.
+You can send messages, fetch, edit and delete messages sent by the Webhook.
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:96]:
```suggestion
Webhooks can send messages to text channels, as well as fetch, edit, and delete their own. These methods are the same for both `Webhook` and `WebhookClient`.
```

File: guide/popular-topics/webhooks.md [hunk 5]
```diff
@@ -94,21 +93,26 @@ webhook.edit({
...
+
+### Sending messages
+
+Webhooks, like bots, can send up to 10 embeds per message. They can also send attachments and normal content. The <DocsLink path="class/Webhook?scrollTo=send">`Webhook#send()`</DocsLink> method to send to a webhook is very similar to the method for sending to a text channel. Webhooks can also choose how the username and avatar will appear when they send the message.
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:100]:
```suggestion
Webhooks, like bots, can send up to 10 embeds per message. They can also send attachments and normal content. The <DocsLink path="class/Webhook?scrollTo=send">`Webhook#send()`</DocsLink> method used to send to a webhook is very similar to the method used for sending to a text channel. Webhooks can also choose how the username and avatar will appear when they send the message.
```

File: guide/popular-topics/webhooks.md [hunk 6]
```diff
@@ -118,28 +122,161 @@ webhookClient.send('Webhook test', {
...
+const client = new Client();
 
-const embed = new Discord.MessageEmbed()
+const embed = new MessageEmbed()
 	.setTitle('Some Title')
 	.setColor('#0099ff');
 
 client.once('ready', async () => {
-	const channel = client.channels.cache.get('222197033908436994');
+	const channel = client.channels.cache.get('1234567890123456');
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:135]:
```suggestion
	const channel = client.channels.cache.get('123456789012345678');
```

File: guide/popular-topics/webhooks.md [hunk 7]
```diff
@@ -118,28 +122,161 @@ webhookClient.send('Webhook test', {
...
-		await webhook.send('Webhook test', {
+		await webhook.send({
+			content: 'Webhook test',
 			username: 'some-username',
 			avatarURL: 'https://i.imgur.com/wSTFkRM.png',
 			embeds: [embed],
 		});
 	} catch (error) {
-		console.error('Error trying to send: ', error);
+		console.error('Error trying to send a message: ', error);
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:147]:
```suggestion
		console.error('Error trying to send a message:', error);
```

File: guide/popular-topics/webhooks.md [hunk 8]
```diff
@@ -118,28 +122,161 @@ webhookClient.send('Webhook test', {
...
+		const webhooks = await channel.fetchWebhooks();
+		const webhook = webhooks.first();
+
+		const message = await webhook.fetchMessage('1234567890123456');
+	} catch (error) {
+		console.error('Error trying to fetch a message: ', error);
+	}
+});
+
+client.login(token);
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:193]:
```suggestion
You can use <DocsLink path="class/Webhook?scrollTo=fetchMessage">`Webhook#fetchMessage()`</DocsLink> to fetch messages previously sent by the Webhook.

```js
const message = await webhookClient.fetchMessage('123456789012345678');
```

File: guide/popular-topics/webhooks.md [hunk 9]
```diff
@@ -118,28 +122,161 @@ webhookClient.send('Webhook test', {
...
+			username: 'some-username',
+			avatarURL: 'https://i.imgur.com/wSTFkRM.png',
+			embeds: [embed],
+		});
+	} catch (error) {
+		console.error('Error trying to edit a message: ', error);
+	}
+});
+
+client.login(token);
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:245]:
```suggestion
You can use <DocsLink path="class/Webhook?scrollTo=editMessage">`Webhook#editMessage()`</DocsLink> to edit messages previously sent by the Webhook.

<!-- eslint-skip -->

```js
const message = await webhook.editMessage('123456789012345678', {
	content: 'Edited!',
	username: 'some-username',
	avatarURL: 'https://i.imgur.com/wSTFkRM.png',
	embeds: [embed],
});
```

File: guide/popular-topics/webhooks.md [hunk 10]
```diff
@@ -118,28 +122,161 @@ webhookClient.send('Webhook test', {
...
+	const channel = client.channels.cache.get('1234567890123456');
+	try {
+		const webhooks = await channel.fetchWebhooks();
+		const webhook = webhooks.first();
+
+		await webhook.deleteMessage('1234567890123456');
+	} catch (error) {
+		console.error('Error trying to delete a message: ', error);
 	}
 });
```
Comment (Danktuary) [inline guide/popular-topics/webhooks.md:282]:
```suggestion
You can use <DocsLink path="class/Webhook?scrollTo=deleteMessage">`Webhook#deleteMessage()`</DocsLink> to delete messages previously sent by the Webhook.

<!-- eslint-skip -->

```js
await webhookClient.deleteMessage('123456789012345678');
```
There's a `client.login()` line below that should be deleted (can't reach it via suggestions)

**Review by Danktuary — CHANGES_REQUESTED** [review]
The examples for fetching/editing/deleting are repetitive and share the same methods whether on a Webhook or WebhookClient. They can all be shortened

**Review by Danktuary — APPROVED** [review]

**Review by almostSouji — APPROVED** [review]

**Pull request merged**